tools/mock-service-host/README.md

```diff
@@ -198,18 +301,187 @@ Feel free to try other commands.
+```diff
+# autorest --version=3.7.3
+           --use=@autorest/go@4.0.0-preview.36 
+           --use=@autorest/gotest@1.3.0 
+           --go 
+           --track2 
+           --output-folder=<output-folder> 
+           --file-prefix="zz_generated_" 
+           --clear-output-folder=false 
+           --go.clear-output-folder=false 
++          --testmodeler.generate-mock-test // remember to add this option if want to run tests
+           --generate-sdk=true 
+           --azure-arm=true 
+           --module-version=0.1.0 
+           --module-name=<module-name>
+           --module=github.com/<organization-name>/<repo-name>/<module-name>
+           <path-to-the-swagger-readme.md>
+# cd <output-folder>
+# go mod tidy && go build && go vet         // resolve dependencies and build
+```
+Now you can try to use the generated lib for mock request.
+
+- Example: agrifood
+
+Below is sample steps for geneartion and mock usage for RP [agrifood](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/agrifood/resource-manager):
+
+```bat
+> autorest --version=3.7.3 
+           --use=@autorest/go@4.0.0-preview.36 
+           --use=@autorest/gotest@1.3.0 
+           --go 
+           --track2 
+           --output-folder=./mock-test-lib/agrifood
+           --file-prefix="zz_generated_" 
+           --clear-output-folder=false 
+           --go.clear-output-folder=false 
+           --testmodeler.generate-mock-test
+           --generate-sdk=true 
+           --azure-arm=true 
+           --module-version=0.1.0 
+           --module-name=agrifood 
+           --module=agrifood 
+           ./azure-rest-api-specs/specification/agrifood/resource-manager/readme.md
+# cd ./mock-test-lib/agrifood        // go to the output folder
+# go mod tidy && go build && go vet  // resolve dependencies, build and install
+# mkdir main && cd main              // create main package for test
+```
+
+Add the following code to the `main.go` file to send `Create Extension` request to Mock Server Host.
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"net/http"
+
+	"agrifood"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"golang.org/x/net/http2"
+)
+
+type MockCredential struct {
+}
+
+func (c *MockCredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (*azcore.AccessToken, error) {
+	return &azcore.AccessToken{Token: "MockToken", ExpiresOn: time.Now().Add(time.Hour * 24).UTC()}, nil
+}
+
+func main() {
+	tr := &http.Transport{}
+	if err := http2.ConfigureTransport(tr); err != nil {
+		fmt.Printf("Failed to configure http2 transport: %v", err)
+	}
+	tr.TLSClientConfig.InsecureSkipVerify = true
+	httpClient := &http.Client{Transport: tr}
+	options := arm.ClientOptions{
+		ClientOptions: policy.ClientOptions{
+			Transport: httpClient,
+		},
+		Endpoint: arm.Endpoint("https://localhost:8443"),
```

Comment:
It is better to provide some context for  'https://localhost:8443'.

Reply:
Described in `# The Behaviour of Mock Service Host` part.